llvm/lib/SYCLLowerIR/ESIMD/ESIMDOptimizeVecArgCallConv.cpp

```diff
@@ -462,9 +462,9 @@ static bool processFunction(Function *F) {
       NewParamTs.push_back(Arg.getType());
       continue;
     }
-    OptimizeableParams.emplace_back(std::move(PI));
+    OptimizeableParams.push_back(PI);
```

Comment:
This std::move caused error on line 467 below: usage of PI after std::move.
FormalParamInfo is small struct, thus there is no need for that kind of optimization using "emplace_back and std::move" here. Keeping this code simpler with push_back will make it more robust.